iceoryx_posh/include/iceoryx_posh/internal/popo/modern_api/base_publisher.inl

```diff
@@ -15,15 +15,15 @@
 #ifndef IOX_POSH_POPO_BASE_PUBLISHER_INL
 #define IOX_POSH_POPO_BASE_PUBLISHER_INL
 
```

Comment:
The new C API is in the root directory. We need some repo refactoring later, but should we have it already symmetric. I.e. this would be in a iceoryx_binding_modern_cpp directory parallel to iceoryx_binding_c ?

Reply:
@elfenpiff @elBoberido @mossmaurice Thoughts ?

Reply:
Yeah, makes sense. As stated in our main Readme we need to split up the CMake target `iceoryx_posh` anyway at some point. I would call it `iceoryx_binding _cpp`.

Reply:
As we have these new API on the master branch now and also the new examples. People who clone this maybe already use it. Is this cool? I'm not sure which stability we reached with the new APIs. Having things more hidden or flagged as "experimental" would be a more conservative way. Or we have to point more to releases and make clear that the master branch is work in progress and maybe unstable

Reply:
I don't think we need to explicitly mention this, it's the same with every project master = unstable, release tag = stable

Reply:
I don't think this API should be moved into its own directory. 
It's not a "binding", it's part of the core iceoryx library.

Reply:
The C binding isn't either, or? Please align with @elfenpiff who made the C "binding". Similar things should have the same naming conventions and should be found in similar places. Like the client libraries in ROS with rclcpp, rclpy and now rclrs for rust

Reply:
I would keep the native API/language within posh or at least not call it binding. When I see a C++ binding, I would wonder in which language iceoryx is written

Reply:
OK fine. Then the modern C++ API will replace old one that is in posh. And the C API is some kind of extra API that will be more like an additional module. Having different APIs side by side would be another option but now understand your thoughts